exporter/exporter.go

```diff
@@ -0,0 +1,180 @@
+package exporter
+
+import (
+	"io"
+	"encoding/json"
+	"strconv"
+
+	"github.com/google/cayley/graph"
+)
+
+type Exporter struct {
```

Comment:
This really looks to me like a job for an interface with a variety of marshalers. Look to [encoding.TextMarshaler](http://golang.org/pkg/encoding/#TextMarshaler) for how you might do this.

Going with a collection of registered interface implementation eases extensibility.